targets/TARGET_STM/serial_api.c

```diff
@@ -393,32 +393,30 @@ void serial_baud(serial_t *obj, int baudrate)
     struct serial_s *obj_s = SERIAL_S(obj);
 
     obj_s->baudrate = baudrate;
```

Comment:
Not related to this PR but maybe put this line only when the init_uart function has returned OK ?